From 3c8240a44dc28a4c7c60c1599799383776f9e6dc Mon Sep 17 00:00:00 2001
From: Lennart Poettering <lennart@poettering.net>
Date: Tue, 31 Aug 2021 10:47:40 +0200
Subject: [PATCH] homed: remove misplaced assert()

(cherry picked from commit 67f9bf897c762ecb61872a5e8e0707a97c8ccd3d)

Conflict:NA
Reference:https://github.com/systemd/systemd/commit/3c8240a44dc28a4c7c60c1599799383776f9e6dc
---
 src/home/homework-mount.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/home/homework-mount.c b/src/home/homework-mount.c
index 5e737687d1..da4f14e08d 100644
--- a/src/home/homework-mount.c
+++ b/src/home/homework-mount.c
@@ -69,9 +69,10 @@ int home_move_mount(const char *user_name_and_realm, const char *target) {
         const char *d;
         int r;
 
-        assert(user_name_and_realm);
         assert(target);
 
+        /* If user_name_and_realm is set, then we'll mount a subdir of the source mount into the host. If
+         * it's NULL we'll move the mount itself */
         if (user_name_and_realm) {
                 subdir = path_join("/run/systemd/user-home-mount/", user_name_and_realm);
                 if (!subdir)
-- 
2.33.0

